### PR TITLE
Include strict-transport-policy response headers

### PR DESCRIPTION
**Pull request by blampe**
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-managed-response-headers-policies.html#managed-response-headers-policies-security

Refs https://github.com/pulumi/home/issues/2490

I'm not sure this will resolve the issue. I suggest we try to repo against get.pulumi-staging.io, deploy this change there, and then confirm/deny the fix works. I'm not familiar with how we typically deploy this -- if there's a better way please let me know!

### CODE REVIEW

**Comment by AaronFriel** [comment]
Moving this to a draft to avoid accidental merges. @blampe is looking into reproducing the issue motivating this.

File: infrastructure/index.ts [hunk 1]
```diff
@@ -222,6 +222,8 @@ const distributionArgs: aws.cloudfront.DistributionArgs = {
         },
         targetOriginId: contentBucket.bucketDomainName.apply(d => `S3-${d}`),
         viewerProtocolPolicy: "redirect-to-https",
+        
+        responseHeadersPolicyId: "67f7725c-6f97-4210-82d7-5512b31e9d03", // SecurityHeadersPolicy
```
Comment (kmosher) [inline infrastructure/index.ts:226]: Should include the link to the AWS docs post here so it's clear this is an AWS thing and not some resource hardcoded from another stack or something.

**Review by kmosher — APPROVED** [review]
I'm in favor of doing this, even if we don't think it's necessary for bug repro, as it hardens against SSL stripping attacks on get.pulumi.com.

**Comment by AaronFriel** [comment]
I agree with @kmosher on hardening against MITM. 

Looking at is a low risk change, and the CI system tests the installer against `pulumi-staging`. I can merge this and validate our install scripts continue to work.

File: infrastructure/index.ts (hunk 1 above)
Comment (AaronFriel) [inline infrastructure/index.ts:227]:
```suggestion
        responseHeadersPolicyId: "67f7725c-6f97-4210-82d7-5512b31e9d03", // SecurityHeadersPolicy from https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-managed-response-headers-policies.html
```

**Comment by AaronFriel** [comment]
Merging this and deploying to staging, will validate `get.pulumi-staging.io`.

**Pull request merged**